### PR TITLE
server: implement distributed engine with rpc

### PR DESCRIPTION
**Pull request by Tricky511**

### CODE REVIEW

File: server/src/lib.rs [hunk 1]
```diff
@@ -70,7 +66,9 @@ impl From<EngineError> for u32 {
         }
     }
 }
-
+lazy_static::lazy_static! {
+    static ref CLIENT: Arc<rpc::client::ClientAsync> = Arc::new(rpc::client::ClientAsync::new());
```
Comment (luanshaotong) [inline server/src/lib.rs:70]: Maybe you can use Box::new()? I remember it is on the heap.

File: server/src/distributed_engine/engine_rpc.rs [hunk 2]
```diff
@@ -1,124 +1,116 @@
...
     use crate::storage_engine::{default_engine::DefaultEngine, StorageEngine};
     use crate::{DistributedEngine, EngineError};
     use common::distribute_hash_table::build_hash_ring;
     use nix::sys::stat::Mode;
     use std::sync::Arc;
     use tokio::time::sleep;
 
+    lazy_static::lazy_static! {
+        static ref CLIENT0: Arc<rpc::client::ClientAsync> = Arc::new(rpc::client::ClientAsync::new());
+        static ref CLIENT1: Arc<rpc::client::ClientAsync> = Arc::new(rpc::client::ClientAsync::new());
```
[the remark this answers is not in the document]
Comment (Tricky511) [reply server/src/distributed_engine/engine_rpc.rs:86]: This is just a test, and in fact there is only one `CLIENT` per server

[the remark this answers is not in the document]
Comment (luanshaotong) [reply server/src/distributed_engine/engine_rpc.rs:86]: Do you mean it test the situation about two clients on different nodes?

[the remark this answers is not in the document]
Comment (luanshaotong) [reply server/src/distributed_engine/engine_rpc.rs:86]: > Do you mean it test the situation about two clients on different nodes?

If so, it's alright.

[the remark this answers is not in the document]
Comment (Tricky511) [reply server/src/distributed_engine/engine_rpc.rs:86]: yes, communication between two servers

File: server/src/lib.rs (hunk 1 above)
Comment (Tricky511) [reply server/src/lib.rs:70]: It should be possible to remove `Arc` directly here, don't care if it's on the heap

Comment (luanshaotong) [reply server/src/lib.rs:70]: @uran0sH Any idea?

Comment (uran0sH) [reply server/src/lib.rs:70]: > It should be possible to remove `Arc` directly here, don't care if it's on the heap

maybe it's ok

**Review by uran0sH — APPROVED** [review]

File: server/src/lib.rs (hunk 1 above)
Comment (luanshaotong) [reply server/src/lib.rs]: @Tricky511 Now you can remove the lazy_static.

**Review by luanshaotong — APPROVED** [review]
LGTM

**Review by uran0sH — APPROVED** [review]

**Pull request merged**